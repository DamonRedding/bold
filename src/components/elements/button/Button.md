# Button

Botões expressam que alguma atividade vai acontecer quando o usuário clicar nele.
São usados para iniciar uma ação, alguns exemplos incluem a ação de Adicionar, Salvar, Deletar.

## Boas práticas

- Não use botões como elementos de navegação. Use Links ao invés,  porque levam o usuário a uma
nova página que não está associada com a ação.
- A label do botão indica ao usuário o que vai acontecer quando ele clicar no botão.
Use somente verbos que descrevem a ação, como Adicionar ou Deletar. As labels devem seguir o
padrão Capitalized (apenas a primeira letra na palavra em maiúsculo).
