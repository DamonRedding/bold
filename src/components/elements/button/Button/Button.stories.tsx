import { boolean, select, text, withKnobs } from '@storybook/addon-knobs'
import { storiesOf } from '@storybook/react'
import * as React from 'react'

import { withPropTypes } from '../../../../stories-addons/withPropTypes'
import { withTheme } from '../../../../stories-addons/withTheme'

import { Button, Type } from './Button'

const typeOptions: {[key in Type]: string} = {
    'normal': 'normal',
    'primary': 'primary',
}

storiesOf('Components', module)
    .addDecorator(withPropTypes(`
    Botões expressam que alguma atividade vai acontecer quando o usuário clicar nele.
    São usados para iniciar uma ação, alguns exemplos incluem a ação de Adicionar, Salvar, Deletar.

    ### BOAS PRÁTICAS:
    - Não use botões como elementos de navegação. Use Links ao invés,  porque levam o usuário a uma
    nova página que não está associada com a ação.
    - A label do botão indica ao usuário o que vai acontecer quando ele clicar no botão.
    Use somente verbos que descrevem a ação, como Adicionar ou Deletar. As labels devem seguir o
    padrão Capitalized (apenas a primeira letra na palavra em maiúsculo).
    `))
    .addDecorator(withKnobs)
    .addDecorator(withTheme())
    .add('Button', () => (
        <Button
            disabled={boolean('disabled', false)}
            type={select('type', typeOptions, 'normal')}
            label={text('label', 'Button')}
            loading={boolean('loading', false)}
        />
    ))
