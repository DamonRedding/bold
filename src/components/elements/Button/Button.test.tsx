import * as React from 'react'
import { shallow, render, mount } from 'enzyme'
import { withTheme } from '../../../test'
import { Button } from './Button'

jest.useFakeTimers()

describe('Button', () => {
    it('deve renderizar corretamente', () => {
        const wrapper = render(withTheme(<Button label='Botão' />))
        expect(wrapper).toMatchSnapshot()
    })

    it('deve ter animação de "carregando" ao especificar onClick com retorno do tipo Promise', () => {
        const delayedFunction = () => {
            return new Promise((resolve, reject) => {
                setTimeout(resolve, 10)
            })
        }

        const wrapper = mount(<Button label='Botão' onClick={delayedFunction} />)
        expect(wrapper.find('button').prop('data-loading')).toBe(false)

        wrapper.simulate('click')
        expect(wrapper.find('button').prop('data-loading')).toBe(true)
    })

    it('não deve ter animação com "onClick" que não seja promise', () => {
        const func = jest.fn()
        const wrapper = mount(<Button label='Botão' onClick={func} />)
        expect(wrapper.find('button').prop('data-loading')).toBe(false)

        wrapper.simulate('click')
        expect(func).toBeCalled()
        expect(wrapper.find('button').prop('data-loading')).toBe(false)
    })

    it('deve adicionar um hint string automaticamente', () => {
        const wrapper = render(withTheme(<Button label='Botão' hint='hint teste' />))
        expect(wrapper).toMatchSnapshot()
    })

    it('deve adicionar um hint custom automaticamente', () => {
        const wrapper = render(withTheme(<Button label='Botão' hint={(<span>hint complexo teste</span>)} />))
        expect(wrapper).toMatchSnapshot()
    })
})
