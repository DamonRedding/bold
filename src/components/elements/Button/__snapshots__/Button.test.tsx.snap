// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button deve adicionar um hint custom automaticamente 1`] = `
.emotion-0 {
  background-color: #ffffff;
  border: 1px solid #b3b3b3;
  border-radius: 2px;
  color: #808080;
  cursor: pointer;
  font-family: "Noto Sans",sans-serif;
  font-size: 0.75rem;
  font-weight: bold;
  line-height: 1.58;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  padding: 0.85rem 2.5rem;
  text-transform: uppercase;
}

.emotion-0:active {
  box-shadow: inset 0 2px 8px 0 #e6e6e6;
}

.emotion-0:disabled {
  opacity: 0.5;
}

.emotion-0:focus {
  border: 1px solid #1e98ff;
  outline: none;
}

<span
  class="hint-wrapper"
>
  <button
    class="emotion-0"
    type="button"
  >
    <span>
      Botão
    </span>
  </button>
</span>
`;

exports[`Button deve adicionar um hint string automaticamente 1`] = `
.emotion-0 {
  background-color: #ffffff;
  border: 1px solid #b3b3b3;
  border-radius: 2px;
  color: #808080;
  cursor: pointer;
  font-family: "Noto Sans",sans-serif;
  font-size: 0.75rem;
  font-weight: bold;
  line-height: 1.58;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  padding: 0.85rem 2.5rem;
  text-transform: uppercase;
}

.emotion-0:active {
  box-shadow: inset 0 2px 8px 0 #e6e6e6;
}

.emotion-0:disabled {
  opacity: 0.5;
}

.emotion-0:focus {
  border: 1px solid #1e98ff;
  outline: none;
}

<span
  class="hint-wrapper"
>
  <button
    class="emotion-0"
    type="button"
  >
    <span>
      Botão
    </span>
  </button>
</span>
`;

exports[`Button deve renderizar corretamente 1`] = `
.emotion-0 {
  background-color: #ffffff;
  border: 1px solid #b3b3b3;
  border-radius: 2px;
  color: #808080;
  cursor: pointer;
  font-family: "Noto Sans",sans-serif;
  font-size: 0.75rem;
  font-weight: bold;
  line-height: 1.58;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  padding: 0.85rem 2.5rem;
  text-transform: uppercase;
}

.emotion-0:active {
  box-shadow: inset 0 2px 8px 0 #e6e6e6;
}

.emotion-0:disabled {
  opacity: 0.5;
}

.emotion-0:focus {
  border: 1px solid #1e98ff;
  outline: none;
}

<button
  class="emotion-0"
  type="button"
>
  <span>
    Botão
  </span>
</button>
`;

exports[`Button deve ter animação de "carregando" ao especificar onClick com retorno do tipo Promise 1`] = `
.emotion-0 {
  background-color: #ffffff;
  border: 1px solid #b3b3b3;
  border-radius: 2px;
  color: #808080;
  cursor: pointer;
  font-family: "Noto Sans",sans-serif;
  font-size: 0.75rem;
  font-weight: bold;
  line-height: 1.58;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  padding: 0.85rem 2.5rem;
  text-transform: uppercase;
}

.emotion-0:active {
  box-shadow: inset 0 2px 8px 0 #e6e6e6;
}

.emotion-0:disabled {
  opacity: 0.5;
}

.emotion-0:focus {
  border: 1px solid #1e98ff;
  outline: none;
}

<ThemeProvider
  theme={
    Object {
      "breakpoint": Object {
        "small": "@media (max-width: 1280px)",
      },
      "color": Object {
        "background": "#f2f2f7",
        "black": "#000000",
        "gray10": "#1a1a1a",
        "gray20": "#333333",
        "gray30": "#4d4d4d",
        "gray40": "#666666",
        "gray50": "#808080",
        "gray60": "#999999",
        "gray70": "#b3b3b3",
        "gray80": "#cccccc",
        "gray90": "#e6e6e6",
        "primary": "#1e98ff",
        "red": "#f6001c",
        "text": "#4d4d4d",
        "white": "#ffffff",
      },
      "font": Object {
        "textFamily": "\\"Noto Sans\\", sans-serif",
        "textSize": 16,
        "titleFamily": "\\"Noto Sans\\", sans-serif",
      },
      "global": Object {
        "@font-face": Object {
          "fontFamily": "\\"Noto Sans\\"",
          "fontStyle": "normal",
          "fontWeight": "400",
          "src": "url(test-file-stub) format(\\"truetype\\")",
        },
      },
    }
  }
>
  <ThemeProvider
    theme={
      Object {
        "breakpoint": Object {
          "small": "@media (max-width: 1280px)",
        },
        "color": Object {
          "background": "#f2f2f7",
          "black": "#000000",
          "gray10": "#1a1a1a",
          "gray20": "#333333",
          "gray30": "#4d4d4d",
          "gray40": "#666666",
          "gray50": "#808080",
          "gray60": "#999999",
          "gray70": "#b3b3b3",
          "gray80": "#cccccc",
          "gray90": "#e6e6e6",
          "primary": "#1e98ff",
          "red": "#f6001c",
          "text": "#4d4d4d",
          "white": "#ffffff",
        },
        "font": Object {
          "textFamily": "\\"Noto Sans\\", sans-serif",
          "textSize": 16,
          "titleFamily": "\\"Noto Sans\\", sans-serif",
        },
        "global": Object {
          "@font-face": Object {
            "fontFamily": "\\"Noto Sans\\"",
            "fontStyle": "normal",
            "fontWeight": "400",
            "src": "url(test-file-stub) format(\\"truetype\\")",
          },
        },
      }
    }
  >
    <Button
      label="Botão"
      onClick={[Function]}
    >
      <WithStyles
        label="Botão"
        onClick={[Function]}
        theme={
          Object {
            "breakpoint": Object {
              "small": "@media (max-width: 1280px)",
            },
            "color": Object {
              "background": "#f2f2f7",
              "black": "#000000",
              "gray10": "#1a1a1a",
              "gray20": "#333333",
              "gray30": "#4d4d4d",
              "gray40": "#666666",
              "gray50": "#808080",
              "gray60": "#999999",
              "gray70": "#b3b3b3",
              "gray80": "#cccccc",
              "gray90": "#e6e6e6",
              "primary": "#1e98ff",
              "red": "#f6001c",
              "text": "#4d4d4d",
              "white": "#ffffff",
            },
            "font": Object {
              "textFamily": "\\"Noto Sans\\", sans-serif",
              "textSize": 16,
              "titleFamily": "\\"Noto Sans\\", sans-serif",
            },
            "global": Object {
              "@font-face": Object {
                "fontFamily": "\\"Noto Sans\\"",
                "fontStyle": "normal",
                "fontWeight": "400",
                "src": "url(test-file-stub) format(\\"truetype\\")",
              },
            },
          }
        }
      >
        <Button
          css={[Function]}
          label="Botão"
          onClick={[Function]}
          theme={
            Object {
              "breakpoint": Object {
                "small": "@media (max-width: 1280px)",
              },
              "color": Object {
                "background": "#f2f2f7",
                "black": "#000000",
                "gray10": "#1a1a1a",
                "gray20": "#333333",
                "gray30": "#4d4d4d",
                "gray40": "#666666",
                "gray50": "#808080",
                "gray60": "#999999",
                "gray70": "#b3b3b3",
                "gray80": "#cccccc",
                "gray90": "#e6e6e6",
                "primary": "#1e98ff",
                "red": "#f6001c",
                "text": "#4d4d4d",
                "white": "#ffffff",
              },
              "font": Object {
                "textFamily": "\\"Noto Sans\\", sans-serif",
                "textSize": 16,
                "titleFamily": "\\"Noto Sans\\", sans-serif",
              },
              "global": Object {
                "@font-face": Object {
                  "fontFamily": "\\"Noto Sans\\"",
                  "fontStyle": "normal",
                  "fontWeight": "400",
                  "src": "url(test-file-stub) format(\\"truetype\\")",
                },
              },
            }
          }
        >
          <Button
            css={[Function]}
            label="Botão"
            onClick={[Function]}
            theme={
              Object {
                "breakpoint": Object {
                  "small": "@media (max-width: 1280px)",
                },
                "color": Object {
                  "background": "#f2f2f7",
                  "black": "#000000",
                  "gray10": "#1a1a1a",
                  "gray20": "#333333",
                  "gray30": "#4d4d4d",
                  "gray40": "#666666",
                  "gray50": "#808080",
                  "gray60": "#999999",
                  "gray70": "#b3b3b3",
                  "gray80": "#cccccc",
                  "gray90": "#e6e6e6",
                  "primary": "#1e98ff",
                  "red": "#f6001c",
                  "text": "#4d4d4d",
                  "white": "#ffffff",
                },
                "font": Object {
                  "textFamily": "\\"Noto Sans\\", sans-serif",
                  "textSize": 16,
                  "titleFamily": "\\"Noto Sans\\", sans-serif",
                },
                "global": Object {
                  "@font-face": Object {
                    "fontFamily": "\\"Noto Sans\\"",
                    "fontStyle": "normal",
                    "fontWeight": "400",
                    "src": "url(test-file-stub) format(\\"truetype\\")",
                  },
                },
              }
            }
            type="normal"
          >
            <button
              className="emotion-0"
              onClick={[Function]}
              onKeyPress={[Function]}
              type="button"
            >
              <span>
                Botão
              </span>
            </button>
          </Button>
        </Button>
      </WithStyles>
    </Button>
  </ThemeProvider>
</ThemeProvider>
`;

exports[`Button deve ter animação de "carregando" ao especificar onClick com retorno do tipo Promise 2`] = `
.emotion-0 {
  background-color: #ffffff;
  border: 1px solid #b3b3b3;
  border-radius: 2px;
  color: #808080;
  cursor: pointer;
  font-family: "Noto Sans",sans-serif;
  font-size: 0.75rem;
  font-weight: bold;
  line-height: 1.58;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  padding: 0.85rem 2.5rem;
  text-transform: uppercase;
}

.emotion-0:active {
  box-shadow: inset 0 2px 8px 0 #e6e6e6;
}

.emotion-0:disabled {
  opacity: 0.5;
}

.emotion-0:focus {
  border: 1px solid #1e98ff;
  outline: none;
}

<ThemeProvider
  theme={
    Object {
      "breakpoint": Object {
        "small": "@media (max-width: 1280px)",
      },
      "color": Object {
        "background": "#f2f2f7",
        "black": "#000000",
        "gray10": "#1a1a1a",
        "gray20": "#333333",
        "gray30": "#4d4d4d",
        "gray40": "#666666",
        "gray50": "#808080",
        "gray60": "#999999",
        "gray70": "#b3b3b3",
        "gray80": "#cccccc",
        "gray90": "#e6e6e6",
        "primary": "#1e98ff",
        "red": "#f6001c",
        "text": "#4d4d4d",
        "white": "#ffffff",
      },
      "font": Object {
        "textFamily": "\\"Noto Sans\\", sans-serif",
        "textSize": 16,
        "titleFamily": "\\"Noto Sans\\", sans-serif",
      },
      "global": Object {
        "@font-face": Object {
          "fontFamily": "\\"Noto Sans\\"",
          "fontStyle": "normal",
          "fontWeight": "400",
          "src": "url(test-file-stub) format(\\"truetype\\")",
        },
      },
    }
  }
>
  <ThemeProvider
    theme={
      Object {
        "breakpoint": Object {
          "small": "@media (max-width: 1280px)",
        },
        "color": Object {
          "background": "#f2f2f7",
          "black": "#000000",
          "gray10": "#1a1a1a",
          "gray20": "#333333",
          "gray30": "#4d4d4d",
          "gray40": "#666666",
          "gray50": "#808080",
          "gray60": "#999999",
          "gray70": "#b3b3b3",
          "gray80": "#cccccc",
          "gray90": "#e6e6e6",
          "primary": "#1e98ff",
          "red": "#f6001c",
          "text": "#4d4d4d",
          "white": "#ffffff",
        },
        "font": Object {
          "textFamily": "\\"Noto Sans\\", sans-serif",
          "textSize": 16,
          "titleFamily": "\\"Noto Sans\\", sans-serif",
        },
        "global": Object {
          "@font-face": Object {
            "fontFamily": "\\"Noto Sans\\"",
            "fontStyle": "normal",
            "fontWeight": "400",
            "src": "url(test-file-stub) format(\\"truetype\\")",
          },
        },
      }
    }
  >
    <Button
      label="Botão"
      onClick={[Function]}
    >
      <WithStyles
        label="Botão"
        onClick={[Function]}
        theme={
          Object {
            "breakpoint": Object {
              "small": "@media (max-width: 1280px)",
            },
            "color": Object {
              "background": "#f2f2f7",
              "black": "#000000",
              "gray10": "#1a1a1a",
              "gray20": "#333333",
              "gray30": "#4d4d4d",
              "gray40": "#666666",
              "gray50": "#808080",
              "gray60": "#999999",
              "gray70": "#b3b3b3",
              "gray80": "#cccccc",
              "gray90": "#e6e6e6",
              "primary": "#1e98ff",
              "red": "#f6001c",
              "text": "#4d4d4d",
              "white": "#ffffff",
            },
            "font": Object {
              "textFamily": "\\"Noto Sans\\", sans-serif",
              "textSize": 16,
              "titleFamily": "\\"Noto Sans\\", sans-serif",
            },
            "global": Object {
              "@font-face": Object {
                "fontFamily": "\\"Noto Sans\\"",
                "fontStyle": "normal",
                "fontWeight": "400",
                "src": "url(test-file-stub) format(\\"truetype\\")",
              },
            },
          }
        }
      >
        <Button
          css={[Function]}
          label="Botão"
          onClick={[Function]}
          theme={
            Object {
              "breakpoint": Object {
                "small": "@media (max-width: 1280px)",
              },
              "color": Object {
                "background": "#f2f2f7",
                "black": "#000000",
                "gray10": "#1a1a1a",
                "gray20": "#333333",
                "gray30": "#4d4d4d",
                "gray40": "#666666",
                "gray50": "#808080",
                "gray60": "#999999",
                "gray70": "#b3b3b3",
                "gray80": "#cccccc",
                "gray90": "#e6e6e6",
                "primary": "#1e98ff",
                "red": "#f6001c",
                "text": "#4d4d4d",
                "white": "#ffffff",
              },
              "font": Object {
                "textFamily": "\\"Noto Sans\\", sans-serif",
                "textSize": 16,
                "titleFamily": "\\"Noto Sans\\", sans-serif",
              },
              "global": Object {
                "@font-face": Object {
                  "fontFamily": "\\"Noto Sans\\"",
                  "fontStyle": "normal",
                  "fontWeight": "400",
                  "src": "url(test-file-stub) format(\\"truetype\\")",
                },
              },
            }
          }
        >
          <Button
            css={[Function]}
            label="Botão"
            onClick={[Function]}
            theme={
              Object {
                "breakpoint": Object {
                  "small": "@media (max-width: 1280px)",
                },
                "color": Object {
                  "background": "#f2f2f7",
                  "black": "#000000",
                  "gray10": "#1a1a1a",
                  "gray20": "#333333",
                  "gray30": "#4d4d4d",
                  "gray40": "#666666",
                  "gray50": "#808080",
                  "gray60": "#999999",
                  "gray70": "#b3b3b3",
                  "gray80": "#cccccc",
                  "gray90": "#e6e6e6",
                  "primary": "#1e98ff",
                  "red": "#f6001c",
                  "text": "#4d4d4d",
                  "white": "#ffffff",
                },
                "font": Object {
                  "textFamily": "\\"Noto Sans\\", sans-serif",
                  "textSize": 16,
                  "titleFamily": "\\"Noto Sans\\", sans-serif",
                },
                "global": Object {
                  "@font-face": Object {
                    "fontFamily": "\\"Noto Sans\\"",
                    "fontStyle": "normal",
                    "fontWeight": "400",
                    "src": "url(test-file-stub) format(\\"truetype\\")",
                  },
                },
              }
            }
            type="normal"
          >
            <button
              className="emotion-0"
              onClick={[Function]}
              onKeyPress={[Function]}
              type="button"
            >
              <span>
                Botão
              </span>
            </button>
          </Button>
        </Button>
      </WithStyles>
    </Button>
  </ThemeProvider>
</ThemeProvider>
`;

exports[`Button não deve ter animação com "onClick" que não seja promise 1`] = `
.emotion-0 {
  background-color: #ffffff;
  border: 1px solid #b3b3b3;
  border-radius: 2px;
  color: #808080;
  cursor: pointer;
  font-family: "Noto Sans",sans-serif;
  font-size: 0.75rem;
  font-weight: bold;
  line-height: 1.58;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  padding: 0.85rem 2.5rem;
  text-transform: uppercase;
}

.emotion-0:active {
  box-shadow: inset 0 2px 8px 0 #e6e6e6;
}

.emotion-0:disabled {
  opacity: 0.5;
}

.emotion-0:focus {
  border: 1px solid #1e98ff;
  outline: none;
}

<ThemeProvider
  theme={
    Object {
      "breakpoint": Object {
        "small": "@media (max-width: 1280px)",
      },
      "color": Object {
        "background": "#f2f2f7",
        "black": "#000000",
        "gray10": "#1a1a1a",
        "gray20": "#333333",
        "gray30": "#4d4d4d",
        "gray40": "#666666",
        "gray50": "#808080",
        "gray60": "#999999",
        "gray70": "#b3b3b3",
        "gray80": "#cccccc",
        "gray90": "#e6e6e6",
        "primary": "#1e98ff",
        "red": "#f6001c",
        "text": "#4d4d4d",
        "white": "#ffffff",
      },
      "font": Object {
        "textFamily": "\\"Noto Sans\\", sans-serif",
        "textSize": 16,
        "titleFamily": "\\"Noto Sans\\", sans-serif",
      },
      "global": Object {
        "@font-face": Object {
          "fontFamily": "\\"Noto Sans\\"",
          "fontStyle": "normal",
          "fontWeight": "400",
          "src": "url(test-file-stub) format(\\"truetype\\")",
        },
      },
    }
  }
>
  <ThemeProvider
    theme={
      Object {
        "breakpoint": Object {
          "small": "@media (max-width: 1280px)",
        },
        "color": Object {
          "background": "#f2f2f7",
          "black": "#000000",
          "gray10": "#1a1a1a",
          "gray20": "#333333",
          "gray30": "#4d4d4d",
          "gray40": "#666666",
          "gray50": "#808080",
          "gray60": "#999999",
          "gray70": "#b3b3b3",
          "gray80": "#cccccc",
          "gray90": "#e6e6e6",
          "primary": "#1e98ff",
          "red": "#f6001c",
          "text": "#4d4d4d",
          "white": "#ffffff",
        },
        "font": Object {
          "textFamily": "\\"Noto Sans\\", sans-serif",
          "textSize": 16,
          "titleFamily": "\\"Noto Sans\\", sans-serif",
        },
        "global": Object {
          "@font-face": Object {
            "fontFamily": "\\"Noto Sans\\"",
            "fontStyle": "normal",
            "fontWeight": "400",
            "src": "url(test-file-stub) format(\\"truetype\\")",
          },
        },
      }
    }
  >
    <Button
      label="Botão"
      onClick={[Function]}
    >
      <WithStyles
        label="Botão"
        onClick={[Function]}
        theme={
          Object {
            "breakpoint": Object {
              "small": "@media (max-width: 1280px)",
            },
            "color": Object {
              "background": "#f2f2f7",
              "black": "#000000",
              "gray10": "#1a1a1a",
              "gray20": "#333333",
              "gray30": "#4d4d4d",
              "gray40": "#666666",
              "gray50": "#808080",
              "gray60": "#999999",
              "gray70": "#b3b3b3",
              "gray80": "#cccccc",
              "gray90": "#e6e6e6",
              "primary": "#1e98ff",
              "red": "#f6001c",
              "text": "#4d4d4d",
              "white": "#ffffff",
            },
            "font": Object {
              "textFamily": "\\"Noto Sans\\", sans-serif",
              "textSize": 16,
              "titleFamily": "\\"Noto Sans\\", sans-serif",
            },
            "global": Object {
              "@font-face": Object {
                "fontFamily": "\\"Noto Sans\\"",
                "fontStyle": "normal",
                "fontWeight": "400",
                "src": "url(test-file-stub) format(\\"truetype\\")",
              },
            },
          }
        }
      >
        <Button
          css={[Function]}
          label="Botão"
          onClick={[Function]}
          theme={
            Object {
              "breakpoint": Object {
                "small": "@media (max-width: 1280px)",
              },
              "color": Object {
                "background": "#f2f2f7",
                "black": "#000000",
                "gray10": "#1a1a1a",
                "gray20": "#333333",
                "gray30": "#4d4d4d",
                "gray40": "#666666",
                "gray50": "#808080",
                "gray60": "#999999",
                "gray70": "#b3b3b3",
                "gray80": "#cccccc",
                "gray90": "#e6e6e6",
                "primary": "#1e98ff",
                "red": "#f6001c",
                "text": "#4d4d4d",
                "white": "#ffffff",
              },
              "font": Object {
                "textFamily": "\\"Noto Sans\\", sans-serif",
                "textSize": 16,
                "titleFamily": "\\"Noto Sans\\", sans-serif",
              },
              "global": Object {
                "@font-face": Object {
                  "fontFamily": "\\"Noto Sans\\"",
                  "fontStyle": "normal",
                  "fontWeight": "400",
                  "src": "url(test-file-stub) format(\\"truetype\\")",
                },
              },
            }
          }
        >
          <Button
            css={[Function]}
            label="Botão"
            onClick={[Function]}
            theme={
              Object {
                "breakpoint": Object {
                  "small": "@media (max-width: 1280px)",
                },
                "color": Object {
                  "background": "#f2f2f7",
                  "black": "#000000",
                  "gray10": "#1a1a1a",
                  "gray20": "#333333",
                  "gray30": "#4d4d4d",
                  "gray40": "#666666",
                  "gray50": "#808080",
                  "gray60": "#999999",
                  "gray70": "#b3b3b3",
                  "gray80": "#cccccc",
                  "gray90": "#e6e6e6",
                  "primary": "#1e98ff",
                  "red": "#f6001c",
                  "text": "#4d4d4d",
                  "white": "#ffffff",
                },
                "font": Object {
                  "textFamily": "\\"Noto Sans\\", sans-serif",
                  "textSize": 16,
                  "titleFamily": "\\"Noto Sans\\", sans-serif",
                },
                "global": Object {
                  "@font-face": Object {
                    "fontFamily": "\\"Noto Sans\\"",
                    "fontStyle": "normal",
                    "fontWeight": "400",
                    "src": "url(test-file-stub) format(\\"truetype\\")",
                  },
                },
              }
            }
            type="normal"
          >
            <button
              className="emotion-0"
              onClick={[Function]}
              onKeyPress={[Function]}
              type="button"
            >
              <span>
                Botão
              </span>
            </button>
          </Button>
        </Button>
      </WithStyles>
    </Button>
  </ThemeProvider>
</ThemeProvider>
`;

exports[`Button não deve ter animação com "onClick" que não seja promise 2`] = `
.emotion-0 {
  background-color: #ffffff;
  border: 1px solid #b3b3b3;
  border-radius: 2px;
  color: #808080;
  cursor: pointer;
  font-family: "Noto Sans",sans-serif;
  font-size: 0.75rem;
  font-weight: bold;
  line-height: 1.58;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  padding: 0.85rem 2.5rem;
  text-transform: uppercase;
}

.emotion-0:active {
  box-shadow: inset 0 2px 8px 0 #e6e6e6;
}

.emotion-0:disabled {
  opacity: 0.5;
}

.emotion-0:focus {
  border: 1px solid #1e98ff;
  outline: none;
}

<ThemeProvider
  theme={
    Object {
      "breakpoint": Object {
        "small": "@media (max-width: 1280px)",
      },
      "color": Object {
        "background": "#f2f2f7",
        "black": "#000000",
        "gray10": "#1a1a1a",
        "gray20": "#333333",
        "gray30": "#4d4d4d",
        "gray40": "#666666",
        "gray50": "#808080",
        "gray60": "#999999",
        "gray70": "#b3b3b3",
        "gray80": "#cccccc",
        "gray90": "#e6e6e6",
        "primary": "#1e98ff",
        "red": "#f6001c",
        "text": "#4d4d4d",
        "white": "#ffffff",
      },
      "font": Object {
        "textFamily": "\\"Noto Sans\\", sans-serif",
        "textSize": 16,
        "titleFamily": "\\"Noto Sans\\", sans-serif",
      },
      "global": Object {
        "@font-face": Object {
          "fontFamily": "\\"Noto Sans\\"",
          "fontStyle": "normal",
          "fontWeight": "400",
          "src": "url(test-file-stub) format(\\"truetype\\")",
        },
      },
    }
  }
>
  <ThemeProvider
    theme={
      Object {
        "breakpoint": Object {
          "small": "@media (max-width: 1280px)",
        },
        "color": Object {
          "background": "#f2f2f7",
          "black": "#000000",
          "gray10": "#1a1a1a",
          "gray20": "#333333",
          "gray30": "#4d4d4d",
          "gray40": "#666666",
          "gray50": "#808080",
          "gray60": "#999999",
          "gray70": "#b3b3b3",
          "gray80": "#cccccc",
          "gray90": "#e6e6e6",
          "primary": "#1e98ff",
          "red": "#f6001c",
          "text": "#4d4d4d",
          "white": "#ffffff",
        },
        "font": Object {
          "textFamily": "\\"Noto Sans\\", sans-serif",
          "textSize": 16,
          "titleFamily": "\\"Noto Sans\\", sans-serif",
        },
        "global": Object {
          "@font-face": Object {
            "fontFamily": "\\"Noto Sans\\"",
            "fontStyle": "normal",
            "fontWeight": "400",
            "src": "url(test-file-stub) format(\\"truetype\\")",
          },
        },
      }
    }
  >
    <Button
      label="Botão"
      onClick={[Function]}
    >
      <WithStyles
        label="Botão"
        onClick={[Function]}
        theme={
          Object {
            "breakpoint": Object {
              "small": "@media (max-width: 1280px)",
            },
            "color": Object {
              "background": "#f2f2f7",
              "black": "#000000",
              "gray10": "#1a1a1a",
              "gray20": "#333333",
              "gray30": "#4d4d4d",
              "gray40": "#666666",
              "gray50": "#808080",
              "gray60": "#999999",
              "gray70": "#b3b3b3",
              "gray80": "#cccccc",
              "gray90": "#e6e6e6",
              "primary": "#1e98ff",
              "red": "#f6001c",
              "text": "#4d4d4d",
              "white": "#ffffff",
            },
            "font": Object {
              "textFamily": "\\"Noto Sans\\", sans-serif",
              "textSize": 16,
              "titleFamily": "\\"Noto Sans\\", sans-serif",
            },
            "global": Object {
              "@font-face": Object {
                "fontFamily": "\\"Noto Sans\\"",
                "fontStyle": "normal",
                "fontWeight": "400",
                "src": "url(test-file-stub) format(\\"truetype\\")",
              },
            },
          }
        }
      >
        <Button
          css={[Function]}
          label="Botão"
          onClick={[Function]}
          theme={
            Object {
              "breakpoint": Object {
                "small": "@media (max-width: 1280px)",
              },
              "color": Object {
                "background": "#f2f2f7",
                "black": "#000000",
                "gray10": "#1a1a1a",
                "gray20": "#333333",
                "gray30": "#4d4d4d",
                "gray40": "#666666",
                "gray50": "#808080",
                "gray60": "#999999",
                "gray70": "#b3b3b3",
                "gray80": "#cccccc",
                "gray90": "#e6e6e6",
                "primary": "#1e98ff",
                "red": "#f6001c",
                "text": "#4d4d4d",
                "white": "#ffffff",
              },
              "font": Object {
                "textFamily": "\\"Noto Sans\\", sans-serif",
                "textSize": 16,
                "titleFamily": "\\"Noto Sans\\", sans-serif",
              },
              "global": Object {
                "@font-face": Object {
                  "fontFamily": "\\"Noto Sans\\"",
                  "fontStyle": "normal",
                  "fontWeight": "400",
                  "src": "url(test-file-stub) format(\\"truetype\\")",
                },
              },
            }
          }
        >
          <Button
            css={[Function]}
            label="Botão"
            onClick={[Function]}
            theme={
              Object {
                "breakpoint": Object {
                  "small": "@media (max-width: 1280px)",
                },
                "color": Object {
                  "background": "#f2f2f7",
                  "black": "#000000",
                  "gray10": "#1a1a1a",
                  "gray20": "#333333",
                  "gray30": "#4d4d4d",
                  "gray40": "#666666",
                  "gray50": "#808080",
                  "gray60": "#999999",
                  "gray70": "#b3b3b3",
                  "gray80": "#cccccc",
                  "gray90": "#e6e6e6",
                  "primary": "#1e98ff",
                  "red": "#f6001c",
                  "text": "#4d4d4d",
                  "white": "#ffffff",
                },
                "font": Object {
                  "textFamily": "\\"Noto Sans\\", sans-serif",
                  "textSize": 16,
                  "titleFamily": "\\"Noto Sans\\", sans-serif",
                },
                "global": Object {
                  "@font-face": Object {
                    "fontFamily": "\\"Noto Sans\\"",
                    "fontStyle": "normal",
                    "fontWeight": "400",
                    "src": "url(test-file-stub) format(\\"truetype\\")",
                  },
                },
              }
            }
            type="normal"
          >
            <button
              className="emotion-0"
              onClick={[Function]}
              onKeyPress={[Function]}
              type="button"
            >
              <span>
                Botão
              </span>
            </button>
          </Button>
        </Button>
      </WithStyles>
    </Button>
  </ThemeProvider>
</ThemeProvider>
`;
